### ISSUE TITLE
Add ability to add single new-line when typing Shift + Enter

### ISSUE THREAD

**Issue by iamthomasbishop** [wordpress-mobile/gutenberg-mobile#559]
## Current Behavior

Currently when I'm focused on a text block (paragraph, heading) and type `Shift + Enter`, a new block is created – which is the same behavior as typing `Enter`.

## Expected Behavior

I expect that if I type `Shift + Enter` in a block, it should simply add a single new-line instead of creating a new block.

### Implementation coverage

- [ ] Android
- [ ] iOS

**Comment by SergioEstevao** [comment]
@iamthomasbishop I tested this on iOS and it works if you apply Shift+Enter in the middle of the paragraph/block.

That said I did found a bug when you try to do this and the end of the paragraph that makes the system crash.

**Comment by iamthomasbishop** [comment]
Hmm, that's not what I'm seeing – maybe this was fixed in a more recent build than the beta I'm testing on (HockeyApp). Here's what I'm seeing, for reference:

![shift_enter_bugs](https://user-images.githubusercontent.com/1200432/53105827-69e9b080-34f7-11e9-8b47-ef5d8be7cd7c.gif)

**Comment by SergioEstevao** [comment]
@iamthomasbishop you are right the Shift+Enter it's only working when using a my laptop keyboard to do it, if I use the device touch keyboard I get the block split.

**Comment by koke** [comment]
@diegoreymendez may confirm but IIRC we tried something similar on aztec and this was impossible to detect on iOS on the screen keyboard

**Comment by diegoreymendez** [comment]
This isn't supported in iOS.  It's only possible through a hardware keyboard.

**Comment by diegoreymendez** [comment]
Sorry, I closed this by mistake and reopened it.

**Comment by iamthomasbishop** [comment]
> This isn't supported in iOS. It's only possible through a hardware keyboard.

@diegoreymendez Fair enough. Just tested on Google Docs and Dropbox Paper and it's the same thing. So we can ax this issue for now if we need to.

**Comment by hypest** [comment]
Shift-Enter with external keyboards would still be useful and perhaps even more when on tablets so, I'd leave this ticket open.

@iamthomasbishop Is it OK with you to just rename it to "... on external keyboards"?

**Comment by SergioEstevao** [comment]
FYI: Shift+Enter using external keyboards works correctly in iOS.

**Comment by hypest** [comment]
Let's find a workaround to enable users to input line breaks without paragraph split. It's really frustrating to do it when one really wants it, to nicely format small paragraphs like the following:

```
    Soft line breaks
    Within a
    Single block

    Followed by
    Another
    Block underneath
```

**Comment by mchowning** [comment]
I noticed something that probably isn't directly helpful but _might_ point the way for some possible solutions...

On both Android and iOS it seems we can get soft-enter-like behavior _sometimes_ if you type very quickly and then hit enter. The video below is from an Android release build not connected to the bundler (so I don't think it was just the result of a bundler delay, although I suspect it may still be related to the asynchronous communication across the bridge). I wasn't able to tell why it would sometimes give me a soft enter and sometimes not.

![soft-new-line mp4](https://user-images.githubusercontent.com/4656348/62374830-d0451700-b50a-11e9-9ff4-063a3895f6f6.gif)

**Comment by mchowning** [comment]
Currently, pressing enter on image captions does create a new line instead of adding a new empty paragraph block. When addressing this, it would be good to give consistent behavior to the image caption as well.

**Comment by designsimply** [comment]
Noting that this came up in Slack and that Eli suggested possibly adding a button to the toolbar as a potential solution.

<sup>(internal reference: p1602473723029700-slack-cellular-happiness)</sup>

**Comment by azollafox** [comment]
From an app review on Google Play

```
Block Editor: Would be great if there was a way to break a line ( ) Because with an Enter on the smartphone keyboard there is always a new block and "Shift + Enter" none of my keyboards know (3 different installed)
```

**Comment by reginabally** [comment]
Another request in an app review on Google Play:

> It would be helpful to have line breaks within the block.

**Comment by reginabally** [comment]
Requested in 3642791-zen.

> As a matter of fact, it is not possible in WP app, to easily begin a new line within a block. Hitting RETURN will begin a new block, and Shift-RETURN as known from former times in simple editors, doesn't work.
Adressing this "feature" before, I was told a workaround by going to HTML-modus and filling in <br> at the wanted position. Which of course is very anoying, doing this lots of times in a posting.

**Comment by mdrockwell** [comment]
3827600-zen

> I can't press enter to go down to the next line without it opening in a separate text block. I'm trying to have two lines of text right beneath each other in the same block.

**Comment by guarani** [comment]
Another request in an app review on Google Play:

> And I hate to see a new block created EVERYTIME I hit enter.

**Comment by guarani** [comment]
And another request in an app review on Google Play:

> Every time I press the Enter key, a new writing box is opened, although I actually only wanted to make a line break instead of starting a new paragraph.

**Comment by mchowning** [comment]
@iamthomasbishop tested and confirmed that shift+enter behavior works for both iPad and iPhones when accessing Gutenberg on the web. Combined with the fact that shift+enter works in the iOS notes app, this gives some hope that maybe something changed and maybe this is now possible. 🤞

**Comment by guarani** [comment]
It looks like it isn't currently possible to support Shift+Enter on iOS, more info below.

> @iamthomasbishop tested and confirmed that shift+enter behavior works for both iPad and iPhones when accessing Gutenberg on the web. Combined with the fact that shift+enter works in the iOS notes app, this gives some hope that maybe something changed and maybe this is now possible. 🤞

After a bit of investigation and collaboration on the team, we found that iOS only provides a public API for Shift key events **when using an external (physical) keyboard** via the `pressesBegan` functions. This API doesn't work with the on-screen virtual keyboard. The iOS Notes app is likely using a private API that we can't use in the WPiOS app (see code [here](https://gist.github.com/koke/3b181a8f7f2c2748b83a3a1d2505facf)) (h/t @koke).

Regarding mobile Safari, using JavaScript it's possible to detect the Shift key being pressed (although it doesn't seem possible to differentiate between it being held and being "on"). This won't help us here though, since Gutenberg Mobile uses something completely different, an Aztec (`UITextView`) instance.

---

An alternative here could be to invert the app's behavior and make soft new-lines the default when pressing Enter. That way, users could add a new Paragraph block when needed, otherwise they can press Enter to create a new line.

**Comment by reginabally** [comment]
Feedback from a user in 4410487-zd-woothemes:

> There’s no reason to make a text block each time you hit Enter!

**Comment by thabotswana** [comment]
3-star iOS app review

>I have tried to use shift+return but it always create new paragraph instead of new line

**Comment by reginabally** [comment]
Another request in 4715088-zd-woothemes:

> Can you tell me how to start a new line rather than a new paragraph when writing text? I am using the Wordpress app on an iPad so shift+enter doesnt work.

**Comment by erricgunawan** [comment]
4745458-zen (translated):

> Please tell me how to start a new line on the edit screen in visual mode.

> If you press the line feed button, the block will change.

**Comment by eduardozulian** [comment]
4860430-zen

User's idea: Hit Enter once for next line, twice for new block.

**Comment by igotdes** [comment]
Also mentioned in #4856871-zen:

> [...] When I am creating a post and writing a textual entry, perhaps in prose, perhaps in verse, if I press return, it always skips a line and creates an irritating space where none is required.

**Comment by reginabally** [comment]
Asked in 4940586-zd-woothemes:

> How do I enter a carriage return (soft return) in a paragraph block. I’m on the iOS app. Is that format not supported?

**Comment by igotdes** [comment]
From [1-star Google Play review](https://play.google.com/console/u/0/developers/7957760354032996428/app/4973361825681449040/user-feedback/review-details?reviewId=gp%3AAOqpTOHUiPFOJRexddxLXsa8xgh_4z237eoSYttNZ7ACUgW7-xiL6sV4kiqOOIwWUMm8v5hTyGIiqjrWefalsQ&corpus=PUBLIC_REVIEWS):

> I do have a problem. Whenever, I want to go to the next line and I click enter, a new paragraph is launched

**Comment by erricgunawan** [comment]
[3-star Google Play review](https://play.google.com/console/u/0/developers/7957760354032996428/app/4973361825681449040/user-feedback/review-details?reviewId=79768b04-cf35-4128-9c62-9db6aa2ba6df&corpus=PUBLIC_REVIEWS):

> The big problem is that I don't want to switch to the next block immediately with Enter on the Android keyboard.

**Comment by rinazrina** [comment]
Another user expecting this works on phone: #37318217-hc

**Comment by SiobhyB** [comment]
From a recent one star review:

> Disastrous experience. I think the developer lacks a basic understanding that in a paragraph, when U seperate the content from its heading that doesn't mean we want to create a totally new block. Imagine a para having its Heading and Content in seperate blocks with wide space between them. Its make their appearance awful. Developer needs to improve that by clicking ENTER key, we don't mean to create extra block but to give the para an effect like that of a content written just under its heading.

**Comment by alinclamba** [comment]
Another user reported this 8567227-zd-a8c
